ckanext/datastore/backend/postgres.py

```diff
@@ -2115,123 +2118,135 @@ def resource_fields(self, id: str) -> dict[str, Any]:
 
         try:
             engine = self._get_read_engine()
-
-            # resource id for deferencing aliases
-            info['meta']['id'] = id
-
-            # count of rows in table
-            meta_sql = sqlalchemy.text(
-                u'SELECT count(_id) FROM "{0}"'.format(id))
-            meta_results = engine.execute(meta_sql)
-            info['meta']['count'] = meta_results.fetchone()[0]  # type: ignore
-
-            # table_type - BASE TABLE, VIEW, FOREIGN TABLE, MATVIEW
-            tabletype_sql = sqlalchemy.text(u'''
-                SELECT table_type FROM INFORMATION_SCHEMA.TABLES
-                WHERE table_name = '{0}'
+            with engine.connect() as conn:
+                conn.execute(_LOCK_TIMEOUT_SQL)
```

Comment:
this should fix the lock we were seeing on ``SELECT count(_id) FROM "{0}`` which is good.